src/App.tsx

```diff
@@ -7,33 +9,48 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+export class App extends Component<{}, State> {
+  state:Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
```

Comment:
remove your event listeners when component will unmount

dont forget that for do it, you need to put your callbacks to variables

Reply:
done